students/mark/Session05/mailroom.py

```diff
@@ -0,0 +1,267 @@
+                    print("debug: adding a function call to add donor.")
+                    add_donor_info(name, donor_db)
+            break
+
+
+def print_menu():
+    getInputVar=(input('''Choose an action:
+
+        1 - Send a Thank You
+        2 - Create a Report
+        3 - Send letters to everyone
+        4 - Quit
+        5 - Add additional contribution
+        '''))
+
+    return getInputVar.strip()
+
+
+def report_sort_key(item):
+    """list.sort(key="<thing>") thing must be a function
+    """
+    return item[1]
+
+
+def print_donor_report(donor_database):
+    """
+    Output report of donors
+    """
+    # First, reduce the raw data into a summary list view
+    report_rows = []
+    for (name, gifts) in donor_database.items():
+        name = format_donor_name(name)
+        total_gifts = sum(gifts)
+        num_gifts = len(gifts)
+        avg_gift = total_gifts / num_gifts
+        report_rows.append((name, total_gifts, num_gifts, avg_gift))
+
+    # sort the report data
+    report_rows.sort(key=report_sort_key)
+    # print it out in with a nice format.
+    print("{:25s} | {:11s} | {:9s} | {:12s}".format(
+          "Donor Name", "Total Given", "Num Gifts", "Average Gift"))
+    print("-" * 66)  # Make 66 little '-'s
+    for row in report_rows:
+        print("{:25s}   {:11.2f}   {:9d}   {:12.2f}".format(*row))
+
+
+def create_letter(donor, donor_db):
+    """
+    Create a thank you letter for the donor
+    :param: donor/string, donor_db dict/string:tuple
+    :returns: string with letter
+    """
+    return '''\n
+          Dear {},
+
+          Thank you for your very kind donation of ${:.2f}.
+          It will be put to very good use.
+          Your generous contributions to date totaling ${:.2f}
+          allow us continue our mission and achive our joint goals.
+
+                         Sincerely,
+                            -The Team
+          '''.format(format_donor_name(donor), donor_db[donor][-1], sum(donor_db[donor]))
+
+
+def send_letters_to_all(donor_db):
+    """
+
+    :params: donor_db/dict,
+    :returns: 0 if completes
+
+    :description:
+    In this version, add a function (and a menu item to invoke it),
+    that goes through all the donors in your donor data structure,
+    generates a thank you letter, and writes it to disk as a text file.
+
+    menu_item, menu #3
+
+    """
+
+    for k in donor_db.keys():
+
+        file_name=format_donor_name(k).replace(',','').replace(" ","_") + ".txt"
+        with open(file_name, 'w') as fh1:
+            fh1.write(create_letter(k, donor_db))
+
+        print(create_letter(k, donor_db))
+
+    return 0
+
+
+def add_donor_info(name, donor_db):
+    """ Add donor info or add a new donor
+    :params: name/string/name of donor db key, donor_db: dictionary of donor names/amts.
+    :return:
+    """
+    if name == "list" or name == "menu":
+        print('Please select a name other than list or menu.')
+        return 15
```

Comment:
This is a strange catch. If you expect them to do list or menu here, you should just let them select list or menu, otherwise I would just assume they won't. Also, you are returning 15, but not ever checking to see if something is returned, so why bother returning anything?